script.js

```diff
@@ -57,668 +57,828 @@
+        clearTimeout(rescuePlugin.UpdateTimer);
+      }
+      return;
+    }
+    var timeCheck = new Date(parseInt(GetCookie("LoginTimeStamp")));
+    var diff = (new Date().getTime() - timeCheck.getTime()) / 1000;
+    if (diff >= 30) {
+      window.onbeforeunload = null;
+      top.location.href = "https://fuelrats.com/get-help";
+      if (rescuePlugin.UpdateTimer !== undefined) {
+        clearTimeout(rescuePlugin.UpdateTimer);
+      }
+      return;
+    }
+    if (GetCookie("sentAnnounce") !== "null") {
+      rescuePlugin.GetInitialRescueInformation(GetCookie("sentAnnounce"));
+      return;
+    }
+    DelCookie("sentAnnounce");
+    if (rescuePlugin.CommanderInfo.CMDRName !== null) {
+      jQuery.ajax({
+        url: rescuePlugin.AnnouncerUrl,
+        type: "POST",
+        dataType: "json",
+        data: {
+          cmdrname: rescuePlugin.CommanderInfo.CMDRName,
+          EO2: rescuePlugin.CommanderInfo.EO2,
+          system: rescuePlugin.CommanderInfo.System,
+          platform: rescuePlugin.CommanderInfo.Platform,
+          extradata: rescuePlugin.CommanderInfo.ExtraData,
+          odyssey: rescuePlugin.CommanderInfo.Odyssey,
+        },
+        success: function () {},
+      });
+    }
+  },
+  BuildLoginForm: function (loadFromOngoingRescue) {
+    var cmdrLabel, contentHolder, topPanel, infoText;
 
-        SetCookie('_cmdrName', rescuePlugin.CommanderInfo.CMDRName, 3600 * 24 * 365);
-        return true;
-    },
-    SendAnnounceToIRC: function () {
-        if (GetCookie('LoginTimeStamp') === 'null') {
-            DelCookie('LoginTimeStamp');
-            window.onbeforeunload = null;
-            top.location.href = 'https://fuelrats.com/get-help';
-            if (rescuePlugin.UpdateTimer !== undefined) {
-                clearTimeout(rescuePlugin.UpdateTimer);
+    if (rescuePlugin.UseClientForm) {
+      cmdrLabel = jQuery('label[for="server_select_nick"]');
+      if (cmdrLabel.length === 0) {
+        setTimeout(function () {
+          rescuePlugin.BuildLoginForm(loadFromOngoingRescue);
+        }, 50);
+        return;
+      }
+      cmdrLabel.text("CMDR Name");
+
+      jQuery(".status").text("Please enter your details below...");
+
+      var systemItem = jQuery(".system");
+      if (systemItem.length === 0) {
+        systemItem = jQuery(
+          `<tr class="system">
+    <td>
+        <label for="server_select_system">System name</label>
+    </td>
+    <td>
+        <input type="text" id="system" required="required" />
+    </td>
+</tr>`
+        );
+        cmdrLabel.parent().parent().after(systemItem);
+      }
+
+      var odysseyItem = jQuery(".odyssey");
+      if (odysseyItem.length === 0) {
+        odysseyItem = jQuery(
+          `<tr class="odyssey">
+    <td>
+        <label for="server_select_odyssey">Odyssey</label>
+    </td>
+    <td>
+        <label style="font-size: .9em;">
+            <input type="checkbox" id="Odyssey" style="width: auto;" value="Odyssey" />
```

Comment:
AFAIK The ID field is case sensitive, so I think this needs to be changed to the suggestion below as otherwise the document.getElementById calls will no succeed.
```suggestion
            <input type="checkbox" id="odyssey" style="width: auto;" value="Odyssey" />
```